experimental/dds/tree2/src/feature-libraries/typed-schema/flexList.ts

```diff
@@ -4,38 +4,83 @@
  */
 
 /**
- * Flexible way to list values.
- * Can be values, functions that return the value (to allow cyclic references to work), or arrays.
+ * A symbol that, when attached to an object as a property and given a truthy value,
+ * causes that object to be treated as an "eager" item in a `FlexList` even if it otherwise would have been considered to be "lazy".
  * @remarks
- * Does not work properly if T can be a function.
+ * Used by `markEager`and `isMarkedEager`.
+ */
+const flexListEager = Symbol("FlexList Eager");
+
+/** Mark the given object as an "eager" item in a `FlexList`, regardless of its type. */
+export function markEager<T>(t: T): T {
+	return isLazy(t)
+		? Object.defineProperty(t, flexListEager, {
+				value: true,
+				configurable: true,
+				enumerable: false,
+				writable: false,
+		  })
+		: t;
+}
+
+/** Check if the given object has been marked "eager" via `markEager`. */
+function isMarkedEager(x: unknown): x is { [flexListEager]: true } {
+	return (x as { [flexListEager]?: boolean })[flexListEager] ?? false;
```

Comment:
If you actually defined an interface with this symbol, you can cast to Partial<MarkedEeager> here and get clearer and simpler code here.

Also I don't see why we need this check seperate from isLazy: might as well just have a single check that handles both wasys to indicate layness vs eager

Reply:
Great idea - I made an interface and combined `isLazy` and `isMarkedEager`.